rir/src/compiler/analysis/dead_store.h

```diff
@@ -178,7 +178,7 @@ class DeadStoreAnalysis {
                 }
             } else if (i->readsEnv()) {
                 observeFullEnv(i->env());
-            } else if (i->exits() || i->effects.contains(Effect::ExecuteCode)) {
+            } else if (i->exits() || i->maxEffect() >= Effect::ExecuteCode) {
```

Comment:
why? we don't treat the effects as ordered anymore. the effects are a set of atomar effects, so it does not make sense the compare them individually. It used to be that e.g. writesEnv would include readsEnv. then it used to make sense to compare individual effects. but now we have sets. so `readsEnv < readsEnv|writesEnv` but `readsEnv <> writesEnv`, they are incomparable.